### PR TITLE
Gjoranv/concurrent metrics retriever

### PR DESCRIPTION
**Pull request by gjoranv**
FYI: @olaaun

### CODE REVIEW

File: metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/NodeMetricsClient.java [hunk 1]
```diff
@@ -41,7 +42,7 @@
     private final HttpClient httpClient;
     private final Clock clock;
 
-    private final Map<ConsumerId, Snapshot> snapshots = new HashMap<>();
+    private final Map<ConsumerId, Snapshot> snapshots = Collections.synchronizedMap(new HashMap<>());
```
Comment (bjorncs) [inline metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/NodeMetricsClient.java]: Use `ConcurrentHashMap` instead. I am pretty sure that `Collections.synchronizedMap` may thrown `ConcurrentModificationException` if the map is iterated on concurrently with mutating operations.

File: metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/ApplicationMetricsRetriever.java [hunk 2]
```diff
@@ -82,8 +82,6 @@ public void deconstruct() {
 
         } catch (Exception e) {
             // Since the task is a ForkJoinTask, we don't need special handling of InterruptedException
-            forkJoinPool.shutdownNow();
```
Comment (bjorncs) [inline metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/ApplicationMetricsRetriever.java]: Consider making `forkJoinPool` final.

**Review by bjorncs — CHANGES_REQUESTED** [review]

File: metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/NodeMetricsClient.java (hunk 1 above)
Comment (gjoranv) [reply metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/NodeMetricsClient.java]: Yes, you are right.

File: metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/ApplicationMetricsRetriever.java (hunk 2 above)
Comment (gjoranv) [reply metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/ApplicationMetricsRetriever.java]: Thanks, I forgot.

**Review by bjorncs — APPROVED** [review]

**Pull request merged**